cmd/embedded-cluster/join.go

```diff
@@ -368,3 +393,100 @@ func runK0sInstallCommand(fullcmd string) error {
 	}
 	return nil
 }
+
+// maybeEnableHA checks if the cluster has more than 3 nodes and prompts the user
+// to enable high availability if it is not already enabled.
+func maybeEnableHA(ctx context.Context) error {
+	kcli, err := kubeutils.KubeClient()
+	if err != nil {
+		return fmt.Errorf("unable to create kube client: %w", err)
+	}
+	embeddedclusterv1beta1.AddToScheme(kcli.Scheme())
+	isHA, err := isHAEnabled(ctx, kcli)
+	if err != nil {
+		return fmt.Errorf("unable to check if HA is enabled: %w", err)
+	}
+	if isHA {
+		return nil
+	}
+	hostname, err := os.Hostname()
+	if err != nil {
+		return fmt.Errorf("unable to get hostname: %w", err)
+	}
+	numNodes, err := waitForNode(ctx, kcli, hostname)
+	if err != nil {
+		return fmt.Errorf("unable to wait for node: %w", err)
+	}
+	if numNodes < 3 {
+		return nil
+	}
+	shouldEnableHA := prompts.New().Confirm("Do you want to enable high availability?", false)
```

Comment:
We should probably add a separate CLI command to be able to enable HA without having to join additional nodes in case someone chooses "No" here by mistake.

Reply:
IMO we can open a story to track that for later. Someone can just remove and re-add the node if that happens, so it's not critical at this stage.

Reply:
Let's discuss this wording too. I don't have a perfect suggestion now, but we should explain this more to a user. Like the fact that you can't go back, what does HA mean, etc.

Reply:
And whenever you're able, I'd like to see a Loom of the output on join since we're changing it a bit.

Reply:
@ajp-io added a loom the PR description.

I will leave out this extra CLI command for now.

Reply:
Copying over a TODO from the loom comments:
> we should discuss how to ask the HA question. The user should know what they're choosing and what the ramifications are (like if you can't go back)